rviz_default_plugins/src/rviz_default_plugins/displays/depth_cloud/depth_cloud_display.cpp

```diff
@@ -200,15 +200,10 @@ void DepthCloudDisplay::setTopic(const QString & topic, const QString & datatype
     depth_transport_property_->setStdString("raw");
     depth_topic_property_->setString(topic);
   } else {
-    int index = topic.lastIndexOf("/");
-    if (index == -1) {
-      return;
-    }
-    QString transport = topic.mid(index + 1);
-    QString base_topic = topic.mid(0, index);
-
-    depth_transport_property_->setString(transport);
-    depth_topic_property_->setString(base_topic);
+    setStatus(
+      rviz_common::properties::StatusProperty::Warn,
+      "Message",
+      "Data type of the topic is not correct. Expect sensor_msgs::msgs::Image");
```

Comment:
It might be nice to include the topic type in the error message:

```suggestion
      "Expected topic type of 'sensor_msgs/msg/Image', saw topic type '" + datatype + "'");
```